Adding initial version of ww-shapemapper WILDS WDL module (#182)
getwilds/wilds-wdl-library: modules/ww-shapemapper/README.md

- **Test workflow**: `testrun.wdl` - Demonstration workflow for testing and examples
- **Documentation**: This README with usage examples and parameter descriptions

## Available Tasks

### `run_shapemapper`

Run ShapeMapper to analyze RNA structure probing data and generate reactivity profiles.

**Inputs:**
- `sample_name` (String): Name identifier for the sample
- `target_fa` (File): FASTA file containing the target RNA sequence(s)
- `modified_r1` (File): R1 FASTQ file from modified (chemically treated) sample
- `modified_r2` (File): R2 FASTQ file from modified (chemically treated) sample
- `untreated_r1` (File): R1 FASTQ file from untreated control sample
- `untreated_r2` (File): R2 FASTQ file from untreated control sample
- `primers_fa` (File, optional): FASTA file containing primer sequences for trimming
- `min_depth` (Int, default=5000): Minimum read depth required for calculating reactivity
- `is_amplicon` (Boolean, default=false): Set to true if data is from amplicon sequencing
- `cpu_cores` (Int, default=2): Number of CPU cores allocated for the task
- `memory_gb` (Int, default=8): Memory allocated for the task in GB

**Outputs:**

> - `output_tar` (File): Compressed tarball of the ShapeMapper output directory

Reviewer: ```suggestion
- `output_tar` (File): Compressed tarball of a folder containing ShapeMapper outputs
```
I would clarify since ShapeMapper does not produce a directory itself

- `log_file` (File): ShapeMapper log file with processing details and quality metrics

## Usage as a Module

### Importing into Your Workflow

> 

Author: That's a great point, the structs are more for batch convenience, not required for task execution. I added a single sample execution example to the README and then clarified that structs and scatter-gather's are useful for scaling, but not specific to ShapeMapper.

3. Paired-end FASTQ files from untreated control samples

## Docker Container

This module uses the `getwilds/shapemapper:2.3` container image, which includes:
- ShapeMapper 2.3 release package with bundled dependencies
- Python 3 runtime environment
- All necessary system dependencies

## Citation

If you use this module in your research, please cite ShapeMapper:

> Busan S, Weeks KM. (2018) Accurate detection of chemical modifications in RNA by mutational profiling (MaP) with ShapeMapper 2. RNA, 24(2):143-148.
> DOI: [10.1261/rna.061945.117](https://doi.org/10.1261/rna.061945.117)

## Contributing

To improve this module or report issues:
1. Fork the [WILDS WDL Library repository](https://github.com/getwilds/wilds-wdl-library)
2. Make your changes following WILDS conventions
3. Test thoroughly with the demonstration workflow
4. Submit a pull request with detailed documentation

## Support and Feedback

For questions about this module or to report issues:
- Open an issue in the [WILDS WDL Library repository](https://github.com/getwilds/wilds-wdl-library/issues)
- Contact the Fred Hutch Data Science Lab at wilds@fredhutch.org

> - See the [WILDS Contributor Guide](https://getwilds.org/guide/) for detailed guidelines

Reviewer: I think this page is outdated (or at least not maintained by us), I would point to the repo's contributing page instead. We should probably take what's relevant to WILDS WDL from this guide and then have the guide point to our contributing page.

Author: Great call, adjusted to point to the WILDS WDL Library Contributing Guide instead in https://github.com/getwilds/wilds-wdl-library/pull/182/commits/ccfc64dc4b10a274ffa1213d8249d667b58b1ce5.
